cli/src/commands/extensions/api.rs

```diff
@@ -380,13 +392,21 @@ async fn parse_lockfile(
 /// possible even after the command has been spawned.
 #[op]
 #[cfg(unix)]
-fn run_sandboxed(process: Process) -> Result<ProcessOutput> {
+fn run_sandboxed(op_state: Rc<RefCell<OpState>>, process: Process) -> Result<ProcessOutput> {
+    let Process { cmd, args, stdin, stdout, stderr, exceptions } = process;
+    let exceptions_strict = exceptions.strict;
+    let resolved_permissions = {
+        let mut state = op_state.borrow_mut();
+        let permissions = state.borrow_mut::<permissions::Permissions>();
+        permissions::Permissions::from(exceptions).subset_of(permissions)
+    }?;
```

Comment:
I think we decided that `run_sanboxed` should always error if `exceptions` is not a strict subset of the extension permissions. So instead of resolving a common subset, this should just be a strict permission check.

That would look something like this:

```suggestion
    let mut state = op_state.borrow_mut();
    let ext_perms = state.borrow_mut::<permissions::Permissions>();
    let child_perms = permissions::Permissions::from(exceptions);
    if !child_perms.subset_of(ext_perms) {
        return Err(anyhow!("Permissions denied by manifest file!"));
    }
```

Or you could have the error message generated by the subset function, which might be able to provide more detail:


```suggestion
    let mut state = op_state.borrow_mut();
    let ext_perms = state.borrow_mut::<permissions::Permissions>();
    let child_perms = permissions::Permissions::from(exceptions);
    child_perms.enforce_subset_of(ext_perms)?;
```

Reply:
When A is subset of B, the common subset is, by definition, A itself. In fact, that's exactly what `subset_of` is returning, rather than a resolution; see [this](https://github.com/phylum-dev/cli/pull/746/files#diff-061d5eeda494ae877735811ed5f387c830d778a7f49a345622fc17ba8f17f0f6R75). In all other cases, there exists one a in A such that for each b in B, b is not prefix of a, which means that A is not subset of B.

The [`paths_subset`](https://github.com/phylum-dev/cli/pull/746/files#diff-061d5eeda494ae877735811ed5f387c830d778a7f49a345622fc17ba8f17f0f6R80) method does indeed return either success (`Ok(())`) or a list of forbidden paths, and that's probably the functionality you are referring to; `subset_of` only takes care of the [additional cases](https://github.com/phylum-dev/cli/pull/746/files#diff-061d5eeda494ae877735811ed5f387c830d778a7f49a345622fc17ba8f17f0f6R69-R73) where at least one of parent and child are boolean permissions, and returns the appropriate permissions set, which could be either the parent's, the child's, or a boolean permission deriving from a combination of those.

I think this is mostly an issue of unclear naming and/or documentation, but the semantics is what we intend.

Reply:
1. You're right, that should probably be split into a separate match with an explicit error.
2. I think it is important that we address this immediately, both in code and in documentation, as it is a public facing API and altering this behavior at a later point would result in a breaking change. I'd strongly suggest supporting the "inherit" use case as it feels like it follows naturally from the distinction we already make between list and boolean permissions.
3. This is a bona fide mistake on my end, and it should definitely result in an error too.

All in all, I think the current overall design makes sense in its intention, that is, centralizing the resolution of permission subsetting in a single testable spot rather than providing a boolean value at the outermost call site.

Reply:
I am happy with this design. We'll just need to settle on the correct way to handle the inherit use case.

> altering this behavior at a later point would result in a breaking change

This is only half true. Changing this case from an error to not an error does not break existing code. That is why I'd rather it be an error for now.

> I'd strongly suggest supporting the "inherit" use case as it feels like it follows naturally from the distinction we already make between list and boolean permissions.

My concern is that there are many different ways implement "inherit" and I'm not certain that `true` is the best option. For example:

* `write: true,`
* `write: PhylumApi.permissions().write,` or more likely `...PhylumApi.permissions()`
* `write: "inherit",`

Reply:
Pinging @cd-work since he may have an opinion on this

Reply:
We discussed in the last sync that `write: true` should not be allowed for now unless `write: true` is also set in the manifest, right? Specifically because it's trivial to change in the future and it might be more confusing than beneficial?

Reply:
My bad then -- must've slipped my mind. In that case, I suggest we ask the user to be explicit with a list for the time being, and we create an issue to add support for a `PhylumApi.permission()` method.